sklearn/impute/_iterative.py

```diff
@@ -747,6 +741,29 @@ def fit_transform(self, X, y=None, **params):
             X, in_fit=True
         )
 
+        err_msg = (
+            "'{}' should be of shape ({},) when an array-like is provided. "
```

Comment:
Sure, I just made another fix. Previously, I decided not to use f-strings in this case because, by design, we need to print a similar error message twice with different contents. With `.format()`, I thought it could be more concise.